validphys2/src/validphys/lhio.py

```diff
@@ -239,6 +232,7 @@ def new_pdf_from_indexes(
         files directly. It is slower and will call LHAPDF to fill the grids,
         but works for sets where the replicas have different grids.
     """
+    import lhapdf
```

Comment:
it could, but I think you need lhapdf for that function (I hadn't really looked at it beyond hiding lhapdf so that it is not imported)

For some stuff you still need lhapdf, I wouldn't fully trust pdfflow.